### ISSUE TITLE
Embedded dashboard- discard button does not work after click on Edit dashboard in embedded dashboard

### ISSUE THREAD

**Issue by umeshme** [apache/superset#30893]
### Bug description

The bug is that

1. Embedd the Superset dashboard. 
2. Click on Edit dashboard button 
3. Click on discard button,

Expected: It should redirect me to the embedded dashboard
Actual: It shows screen which keeps on loading

 Logs does not show any error. 

In network trace after this GET call, nothing happens
embedded/412a5023-617f-435b-9e08-2e821b581902

In case if anyone came across similar issues, please advice.

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by jamiemac87** [comment]
Similar issue with version `4.1.1`. 

**Repro Steps:**
1. Make guest role Admin
2. embed dashboard
3. Click on "edit dashboard" within title
4. click "discard" button

Results in following error
```
{"errors": [{"message": "403 Forbidden: You don't have the permission to access the requested resource. It is either read-protected or not readable by the server.", "error_type": "GENERIC_BACKEND_ERROR", "level": "error", "extra": {"issue_codes": [{"code": 1011, "message": "Issue 1011 - Superset encountered an unexpected error."}]}}]}
```

**Comment by rusackas** [comment]
@Vitor-Avila might know better than me, but I didn't think we actually supported edit mode in embedded dashboards?

**Comment by Vitor-Avila** [comment]
@rusackas technically what's visible/"possible" in embedded mode depends on the permissions assigned to the role set for the guest role. In reality, I think embedded was mostly built for view mode, so I don't think this is expected to work.

**Comment by rusackas** [comment]
Agreed... the functionality is not intended to work. We're open to PRs that somehow properly detect embedded mode and disable the appropriate button(s), of course. A means to determine embedded state might help us disable a _bunch_ of things that shouldn't be happening/supported in embedded dashboards, actually.
